src/storage/on_disk.rs

```diff
@@ -0,0 +1,113 @@
+use std::path::Path;
+
+use alloy_primitives::{keccak256, Address, Bytes, FixedBytes, B256, B64, U256};
+use libmdbx::{Database, DatabaseOptions, NoWriteMap};
+use revm::primitives::Bytecode;
+
+use super::{AccountBasic, EvmCode, Storage};
+
+/// A storage that stores chain data in a MDBX database.
+#[derive(Debug)]
+pub struct OnDiskStorage {
+    inner: Database<NoWriteMap>,
+}
+
+impl OnDiskStorage {
+    /// Opens the on-disk storage at the specified path.
+    pub fn open(path: impl AsRef<Path>) -> Result<Self, libmdbx::Error> {
+        let db = Database::open_with_options(
+            &path,
+            DatabaseOptions {
+                max_tables: Some(16),
+                ..DatabaseOptions::default()
+            },
+        )?;
+        Ok(Self { inner: db })
+    }
+}
+
+impl Storage for OnDiskStorage {
+    type Error = libmdbx::Error;
+
+    fn basic(
+        &self,
+        address: &alloy_primitives::Address,
+    ) -> Result<Option<super::AccountBasic>, Self::Error> {
+        let tx = self.inner.begin_ro_txn()?;
+        let Some(balance) = tx
+            .open_table(Some("balance"))
```

Comment:
Is it slow to call `open_table` on every read?

Reply:
We need a benchmark to answer this question.